app/javascript/packs/tta-report.js

```diff
@@ -0,0 +1,35 @@
+const openForm = function () {
+  const form = document.querySelector("#tta-activity-form")
+  form.classList.remove("display-none") 
+}
+
+const closeForm = function () {
+  const input = document.querySelector("#tta-report-id")
+  input.value = ""
+  const form = document.querySelector("#tta-activity-form")
+  form.classList.add("display-none")
+}
+
+document.querySelector("#js-add-tta-report").addEventListener("keydown", event => {
+  if (event.keyCode == 32) {
```

Comment:
as per https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#accessibility_concerns

```
Warning: Be careful when marking up links with the button role. Buttons are expected to be triggered using the Space or Enter key, while links are expected to be triggered using the Enter key. In other words, when links are used to behave like buttons, adding role="button" alone is not sufficient. It will also be necessary to add a key event handler that listens for the Space key in order to be consistent with native buttons.

```